tests/common_docker.py

```diff
@@ -40,19 +40,26 @@ def docker_compose_cmd(arg_list, use_common_files=True):
         for file in COMPOSE_FILES + extra_files:
             files_args += " -f %s" % file
 
-    subprocess.check_call("docker-compose -p %s %s %s" % (conftest.docker_compose_instance, files_args, arg_list), shell=True)
+    cmd = "docker-compose -p %s %s %s" % (conftest.docker_compose_instance, files_args, arg_list)
+    subprocess.check_call(cmd, shell=True)
 
 
 def stop_docker_compose():
     # take down all COMPOSE_FILES and the s3 specific files
     docker_compose_cmd(" -f ../docker-compose.storage.s3.yml -f ../extra/travis-testing/s3.yml down -v")
-
     set_setup_type(None)
 
 
 def start_docker_compose():
     inline_logs = pytest.config.getoption("--inline-logs")
+
+    while os.path.isfile(conftest.lock_file):
+        logging.debug("waiting for lock")
+        time.sleep(1)
+
+    os.mknod(conftest.lock_file)
     docker_compose_cmd("up -d")
+    os.remove(conftest.lock_file)
```

Comment:
Several things:

* Don't do this here. Instead, move it into the `docker_compose_cmd` function, so that all commands are covered. Tearing it down is likely to be equally buggy.
* Wrap the docker-compose command in a `try`, and make sure that the `os.remove` happens inside the `finally` of that `try`.
* The whole thing has a race condition: If the file is created between your test and the file creation, the invocation will simply fail. The correct way to do this is to attempt the file creation immediately, and then test for AlreadyExists exception and do the sleep there. Wrap the whole thing in a loop which you only exit if the call is successful.